hazelcast-jet-core/src/main/java/com/hazelcast/jet/core/AbstractProcessor.java

```diff
@@ -272,6 +272,14 @@ protected boolean tryProcess4(@Nonnull Object item) throws Exception {
      * The default implementation just emits the watermark downstream
      * to all ordinals.
      * <p>
+     * <i>Caution for jobs with at-least-once guarantee</i><br>
+     * In snapshotted jobs with <i>at least once</i> processing guarantee it
```

Comment:
we should be consistent with usage of "at-least-once" vs "at least once" and "exactly-once". I think the one with dash is the correct usage.